axelrod/match_generator.py

```diff
@@ -167,3 +167,61 @@ def estimated_size(self):
         """Rough estimate of the number of matches that will be generated."""
         size = self.__len__() * (1. / self.prob_end) * self.repetitions
         return size
+
+class SpatialMatches(RoundRobinMatches):
+
+    def __init__(self, players, turns, game, repetitions, edges):
+
+        player_indices = list(range(len(players)))
+        node_indices = sorted(set([node for edge in edges for node in edge]))
+        if player_indices != node_indices:
+            raise ValueError("The graph edges do not include all players.")
+
+        self.edges = edges
+        super(SpatialMatches, self).__init__(players, turns, game, repetitions)
+        """
+        A class that generates spatial structure matches. In spatial structure
+        matches the players do not play against every other player, instead they
+        only play their neighbors.
+
+        In literature the authors interpreted spatial tournaments topology
+        as a 2D square lattice. We want to create a class that will not take as
+        an argument only a 2D lattice but any given graph.
+
+        After considering various python packages for graphs, we ended up using
+        a simple dictionary that includes the edges.So the players - nodes will
+        only play players that are connected to with an edge.
+
+        We have chosen to implement spatial structure tournaments because
+        it has spon various tournaments after the work of Nowak and May 1992,
+        <http://www.nature.com/nature/journal/v359/n6398/abs/359826a0.html>
+        which proved that in PD cooperation behavior can merge for spatial
+        games.
+
+        Moreover, IPD tournaments with this topology have been conducted by
+        various researchers such as Lindgren and Nordahl in 1994
+        <http://www.sciencedirect.com/science/article/pii/0167278994902895>
+        and Brauchli, Killingback and Doebeli 1999.
+        <http://www.zoology.ubc.ca/~doebeli/reprints/Doe38.pdf>
+
+        Parameters
+        ----------
+        players : list
+            A list of axelrod.Player objects
+        turns : integer
+            The number of turns per match
+        game : axelrod.Game
+            The game object used to score the match
+        repetitions : int
+            The number of repetitions of a given match
+        edges : dictionary
```

Comment:
Can you explain what this dictionary looks like? In the tests below it looks like a list.